### PR TITLE
Stabilize Lexer from rust/lex as F/lex

### PR DESCRIPTION
**Pull request by adjivas**
Proposition of resolution for #4 and  #11's issue.
Thank you yet for the help @ulysseB.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -86,6 +86,8 @@ cuda = []
 default = []
 format_exh = []
 mppa = []
+lex = []
```
Comment (ulysseB) [inline Cargo.toml]: Why add a `lex` feature ?

File: telamon-gen/build.rs [hunk 2]
```diff
@@ -1,9 +1,32 @@
+extern crate cc;
 extern crate lalrpop;
 
+
 /// Adds a dependency to the build script.
 fn add_dependency(dep: &str) { println!("cargo:rerun-if-changed={}", dep); }
 
 fn main() {
+    // Compile the lexer.(`LEX="flex" cargo build --features "lex"`)
+    #[cfg(feature = "lex")]
```
Comment (ulysseB) [inline telamon-gen/build.rs]: Why do this only when the lex feature is activated ? Couldn't we always do this ?

File: telamon-gen/src/exh.l [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+equal "="
+divide "/"
+
+choice_ident [a-z][a-z_0-9]*
+value_ident [A-Z][A-Z_0-9]*
+set_ident [A-Z][A-Za-z_0-9]*
+var \${alpha_num}+
+code \"[^\n\"]*\"
+
+/* https://github.com/a-haas/Compilation/blob/6fe46a65d64decdb648feba790bb7b1d2980ca0a/yacc/setparser.l */
```
Comment (ulysseB) [inline telamon-gen/src/exh.l]: What are these comment ? Should they stay here ?

File: telamon-gen/build.rs (hunk 2 above)
Comment (adjivas) [reply telamon-gen/build.rs]: That not needed and will always require the f/lex dependency.
That why this feature is only here for regenerate the C file lexer if specified.

**Pull request merged**